cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/types/TupleType.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2023, Fraunhofer AISEC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *                    $$$$$$\  $$$$$$$\   $$$$$$\
+ *                   $$  __$$\ $$  __$$\ $$  __$$\
+ *                   $$ /  \__|$$ |  $$ |$$ /  \__|
+ *                   $$ |      $$$$$$$  |$$ |$$$$\
+ *                   $$ |      $$  ____/ $$ |\_$$ |
+ *                   $$ |  $$\ $$ |      $$ |  $$ |
+ *                   \$$$$$   |$$ |      \$$$$$   |
+ *                    \______/ \__|       \______/
+ *
+ */
+package de.fraunhofer.aisec.cpg.graph.types
+
+import de.fraunhofer.aisec.cpg.graph.Name
+
+/**
+ * Represents a tuple of types. Primarily used in resolving function calls with multiple return
+ * values.
+ */
+class TupleType(types: List<Type>) : Type() {
```

Comment:
So, a `TupleType` is not an `ObjectType` which "is the main type in the Type system. ObjectTypes describe objects, as instances of a class. This also includes primitive data types."?

Reply:
Is there anything to gain by subclassing object type? It does not describe a single object's type but is the type of potentially multiple objects. Not quite sure.

Reply:
If I understand that (undocumented) function correctly, it is only ever true for object types with generics; which the tuple type cannot have anyway. A singular element of a tuple type yes, but not the tuple type its self. So I guess we should be fine. Fingers crossed.

Reply:
ok